lib/project-data.ts

```diff
@@ -39,11 +41,62 @@ export class ProjectData implements IProjectData {
 								suppressCommandHelp: true},
 							this.projectFilePath, err.toString());
 					}
-				}
+
+					if(data) {
+						this.projectId = data.id;
+					} else { // This is the case when we have package.json file but nativescipt key is not presented in it
+						this.tryToUpgradeProject().wait();
+					}
+				} 
+			} else { // This is the case when no project file found
+				this.tryToUpgradeProject().wait();
+			}
+		}).future<void>()();
+	}
+
+	private throwNoProjectFoundError(): void {
+		this.$errors.fail("No project found at or above '%s' and neither was a --path specified.", process.cwd());
```

Comment:
This produces wrong error message when --path is specified. The same mistake is present in common lib: https://github.com/telerik/mobile-cli-lib/pull/329/files#diff-2fc55d29fa5bf8690e0a4ae30a71effdL32